modules/smtp/modules.php

```diff
@@ -1893,7 +1891,9 @@ function save_imap_draft($atts, $id, $session, $mod, $mod_cache, $uploaded_files
 
     foreach ($mailbox_page[1] as $mail) {
         $msg_header = $imap->get_message_headers($mail['uid']);
-        if ($msg_header['Message-Id'] === $mime->get_headers()['Message-Id']) {
+        //stalwart return Message-ID instead of Message-Id and we add trim to avoid any extra spaces
+        $msg_header_id = $msg_header['Message-ID'] ? trim($msg_header['Message-ID']) : trim($msg_header['Message-Id']);
```

Comment:
I suggest we do a strtolower on all headers before using them - their case combinations might be many. Also, the way you are checking it will generate a php notice that Message-ID key does not exist in the array. You should check with isset or empty.